From 5166f3dd04980177e46293408d998087040d4c19 Mon Sep 17 00:00:00 2001
From: William D C M SILVA <colen@apache.org>
Date: Tue, 24 Jan 2017 10:06:49 -0200
Subject: [PATCH] OPENNLP-955: Fixes docbegin feature generator

The docbegin was mapped to PreviousMapFeatureGenerator instead of DocumentBeginFeatureGenerator. Also changed the name of the factory by appending the sufix.

Fixes OPENNLP-955
Closes apache/opennlp#84
---
 .../opennlp/tools/util/featuregen/GeneratorFactory.java   | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/opennlp-tools/src/main/java/opennlp/tools/util/featuregen/GeneratorFactory.java b/opennlp-tools/src/main/java/opennlp/tools/util/featuregen/GeneratorFactory.java
index bccbea23a..fa97f43a6 100644
--- a/opennlp-tools/src/main/java/opennlp/tools/util/featuregen/GeneratorFactory.java
+++ b/opennlp-tools/src/main/java/opennlp/tools/util/featuregen/GeneratorFactory.java
@@ -259,15 +259,15 @@ static void register(Map<String, XmlFeatureGeneratorFactory> factoryMap) {
     }
   }
 
-  static class DocumentBeginFeatureGenerator implements XmlFeatureGeneratorFactory {
+  static class DocumentBeginFeatureGeneratorFactory implements XmlFeatureGeneratorFactory {
 
     public AdaptiveFeatureGenerator create(Element generatorElement,
         FeatureGeneratorResourceProvider resourceManager) {
-      return new PreviousMapFeatureGenerator();
+      return new DocumentBeginFeatureGenerator();
     }
 
     static void register(Map<String, XmlFeatureGeneratorFactory> factoryMap) {
-      factoryMap.put("docbegin", new DocumentBeginFeatureGenerator());
+      factoryMap.put("docbegin", new DocumentBeginFeatureGeneratorFactory());
     }
   }
 
@@ -633,7 +633,7 @@ static void register(Map<String, XmlFeatureGeneratorFactory> factoryMap) {
     CharacterNgramFeatureGeneratorFactory.register(factories);
     DefinitionFeatureGeneratorFactory.register(factories);
     DictionaryFeatureGeneratorFactory.register(factories);
-    DocumentBeginFeatureGenerator.register(factories);
+    DocumentBeginFeatureGeneratorFactory.register(factories);
     PreviousMapFeatureGeneratorFactory.register(factories);
     SentenceFeatureGeneratorFactory.register(factories);
     TokenClassFeatureGeneratorFactory.register(factories);
